plugins/outputs/timestream/timestream.go

```diff
@@ -113,6 +119,10 @@ func (t *Timestream) Connect() error {
 		if t.SingleTableDimensionNameForTelegrafMeasurementName != "" {
 			return fmt.Errorf("in '%s' mapping mode, do not specify SingleTableDimensionNameForTelegrafMeasurementName key", MappingModeMultiTable)
 		}
+
+		if t.UseMultiMeasureRecords && t.MeasureNameForMultiMeasureRecords == "" {
+			return fmt.Errorf("in '%s' mapping mode, with multi-measure enabled, key MeasureNameForMultiMeasureRecords is required", MappingModeMultiTable)
+		}
```

Comment:
Why is this different from the code in lines 104 to 110?

Reply:
when ingesting in MappingModeSingleTable and UseMultiMeasureRecords is enabled, we use measurementName ( from line protocol ) as multiMeasure name in timestream, where as in MappingModeMultiTable mode ( data is ingested to multiple tables ) and UseMultiMeasureRecords is enabled, we ask for measureName(MeasureNameForMultiMeasureRecords) that needs to be used, this is due to the fact that measurementName(line protocol) is used as timestream table name. Hope that clarifies, Readme has some details about different combinations for these configs.

Reply:
@nirmeshk Please add any details I might have missed here.

Reply:
Fair enough. It would be good if you can add this explanation as a comment to the code...

Reply:
The comment is still missing.